File: python/sumOfAny3Nos.py
def sumofTwoEqualsX(arr, x):
    first = 0
    last = len(arr)-1
    while(first<last):
        print arr[first], " + ", arr[last], " = ", arr[first]+arr[last]
        if arr[first]+arr[last]==x:
            return True
        elif arr[first]+arr[last]<x:
            first+=1
        else:
            last-=1
    return False

def sumOfAny3NosEqualsZero(arr):
    for i in range(len(arr)):
        print arr[i]
        newArr = arr[:i] + arr[i+1:]
        print newArr
        if sumofTwoEqualsX(newArr, -arr[i])==True:
            return True

def main():
    arr = [-5,-4,-3,4,5,6]
    if sumOfAny3NosEqualsZero(arr):
        print "Yes"
    else:
        print "No"
    return 0

if __name__ == "__main__":
    main()

File: c/sumOfAny3Nos.c
#include <stdio.h>
#include <stdbool.h>

bool sumofTwoEqualsX(int *arr, int x, int except, int last){
  int first = 0;
  while(first<last){
    if(first!=except && last!=except){
      printf("\t{%d, %d} = %d\n", arr[first], arr[last], (arr[first]+arr[last]) );
      if((arr[first]+arr[last])==x){
        return true;
      }else if(arr[first]+arr[last]<x){
        first++;
      }else{
        last--;
      }
    }else if(first==except){
      first++;
    }else{
      last--;
    }
  }
  return false;
}

bool sumOfAny3NosEqualsZero(int *arr, int len){
  int i = 0;
  for(i=0; i<len; i++){
    printf("%d ", arr[i]);
    if(sumofTwoEqualsX( arr, -(arr[i]), i, len-1)==true){
      return true;
    }
  }
  return false;
}

int main(){
  int arr[] = {-5,-4,-3,4,5,6};
  int len = sizeof(arr)/sizeof(int);
  if(sumOfAny3NosEqualsZero(arr, len)==true){
    printf("Yes\n");
  }else{
    printf("No\n");
  }
  return 0;
}
